### ISSUE TITLE
parity sync catches up too slow (32gb RAM, SSD, fast internet, 6 cores)

### ISSUE THREAD

**Issue by 1blockologist** [paritytech/parity-ethereum#11471]
- **Parity Ethereum version**: v2.7.2-stable
- **Operating system**: Linux
- **Installation**: one-line installer (bash <(curl https://get.parity.io -L))
- **Fully synchronized**: yes
- **Network**: ethereum
- **Restarted**: yes

I am using Parity 2.7.2 on a virgin compute instance, and after the snapshots were downloaded and it gets to the blocks, it is about 33,000 blocks behind. I came back a few hours to compare and it is 32,300 blocks behind. This will take too long to catch up at this rate.

I am running with no flags. I plan to use this node to get and submit work to the mainnet network.

Are there flags I can use these days to get this synced much faster? I know fast is a default flag now so a lot of online information is 3 years old and outdated. Maybe there is a still relevant flag for a cache or maybe data I can do without or make it leverage more RAM and CPU?

Also one peculiar thing is that it seems to download some blocks multiple times, example:
```
2020-02-09 17:27:33 UTC Syncing #9417732 0xe0a5…9333     0.20 blk/s   17.4 tx/s    2.0 Mgas/s      0+ 1856 Qed  #9419590   37/50 peers    621 KiB chain   73 MiB db  110 MiB queue  572 MiB sync  RPC:  0 conn,    0 req/s,    0 µs
2020-02-09 17:27:38 UTC Syncing #9417732 0xe0a5…9333     0.00 blk/s    0.0 tx/s    0.0 Mgas/s      0+ 1856 Qed  #9419590   37/50 peers    621 KiB chain   73 MiB db  110 MiB queue  572 MiB sync  RPC:  0 conn,    0 req/s,    0 µs
```

Those 5 seconds also could have been used to catch up to the top of the chain faster.  I was thinking these are records of uncles or orphans, but I'm thrown off by the same block hash or maybe thats how it works. This seems to exacerbate the problem.

**Comment by dvdplm** [comment]
The main bottleneck to syncing is IO. Could it be that the disks used with the system are underpowered? I’m afraid there is no magic flag to enable, but 0.2blk/s seems slow, slower than my laptop for example.

**Comment by 1blockologist** [comment]
```
sync; dd if=/dev/zero of=speedtest.txt bs=1M count=1024; rm speedtest.txt
1024+0 records in
1024+0 records out
1073741824 bytes (1.1 GB, 1.0 GiB) copied, 0.931433 s, 1.2 GB/s
```

The compute instance acts like an SSD @dvdplm

**Comment by rlinxy** [comment]
same here, extremely slow. Version 2.5.x used to be OK, took around 1 day to sycn ETC blocks. For version 2.7.2, it's has been syncing 3 days and it's still 300,000 blocks behind.

**Comment by dvdplm** [comment]
> dd if=/dev/zero of=speedtest.txt bs=1M count=1024; rm speedtest.txt

Seems like it should be fast enough indeed.

The reason for slow imports can be tricky to chase down: network, cpu, ram, particularly big blocks etc. VPCs can have slowdowns out of sight of the normal user when e.g. the virtual instance is moved between hosts.

> Those 5 seconds also could have been used to catch up to the top of the chain faster. I was thinking these are records of uncles or orphans, but I'm thrown off by the same block hash or maybe thats how it works. This seems to exacerbate the problem.

No, I think you're misinterpreting the data you're seeing there. Take a look at https://wiki.parity.io/FAQ#what-does-paritys-command-line-output-mean for a detailed description; the application outputs logs every 5sec but that doesn't mean it's blocked on any particular block. There are many threads doing work in parallel, some downloading, some verifying blocks, others executing RPC calls etc. The log out put (its own thread actually) just prints a summary of the current status.

**Comment by 1blockologist** [comment]
Yeah I worked with my VPC provider to determine that it is impossible. Geth works. Do not recommend.

**Comment by rlinxy** [comment]
Why closed it, this issue has not been solved yet. As I mentioned, same VPS gets ETC synced within 2 days by using version 2.5.x, much more faster than version 2.7.2

**Comment by dvdplm** [comment]
> Why closed it, this issue has not been solved yet. As I mentioned, same VPS gets ETC synced within 2 days by using version 2.5.x, much more faster than version 2.7.2

The original reporter closed the issue; not sure why tbh. 

Are you seeing the same odd behavior on 2.5 vs 2.7 on other networks or is this ETC specific?

**Comment by 1blockologist** [comment]
@dvdplm @rlinxy Parity has a history of not considering these kinds of things issues. Have a HDD? get lost, have a system that's not meant to be syncing 100% of the time? Not for you. It wouldn't be surprising to me that the team would form consensus that VPC's sharing resources aren't their ideal node either. 

I didn't know about your observation that this problem isn't showing on other EVM chains. Thats pretty interesting

**Comment by dvdplm** [comment]
@1blockologist your comment is inaccurate, irrelevant and rude and that kind of behaviour is not acceptable here. Please keep the conversations here focused on technical aspects; I'm sure you can find other venues to vent your frustration and opinions.

**Comment by 1blockologist** [comment]
It's the accurate reason about why I closed the question though, nothing more nothing less. Its just preemptive issue management on an issue I opened myself, based on prior, valid experiences, which mirror's the community and collective consciousness about the parity software.

**Comment by dvdplm** [comment]
@rlinxy Please open a new issue, closing this now.
